app/src/ui/cherry-pick/choose-target-branch.tsx

```diff
@@ -0,0 +1,173 @@
+import { BranchList, IBranchListItem, renderDefaultBranch } from '../branches'
+import { ClickSource } from '../lib/list'
+
+interface IChooseTargetBranchDialogProps {
+  /**
+   * See IBranchesState.defaultBranch
+   */
+  readonly defaultBranch: Branch | null
+
+  /**
+   * The currently checked out branch
+   */
+  readonly currentBranch: Branch
+
+  /**
+   * See IBranchesState.allBranches
+   */
+  readonly allBranches: ReadonlyArray<Branch>
+
+  /**
+   * See IBranchesState.recentBranches
+   */
+  readonly recentBranches: ReadonlyArray<Branch>
+
+  /**
+   * A function that's called when the user selects a branch and hits start
+   * cherry pick
+   */
+  readonly onCherryPick: (targetBranch: Branch) => void
+
+  /**
+   * A function that's called when the dialog is dismissed by the user in the
+   * ways described in the Dialog component's dismissable prop.
+   */
+  readonly onDismissed: () => void
+}
+
+interface IChooseTargetBranchDialogState {
+  /** The currently selected branch. */
+  readonly selectedBranch: Branch | null
+
+  /** The filter text to use in the branch selector */
+  readonly filterText: string
+}
+
+/** A component for initiating a rebase of the current branch. */
+export class ChooseTargetBranchDialog extends React.Component<
+  IChooseTargetBranchDialogProps,
+  IChooseTargetBranchDialogState
+> {
+  public constructor(props: IChooseTargetBranchDialogProps) {
+    super(props)
+
+    this.state = {
+      selectedBranch: null,
+      filterText: '',
+    }
+  }
+
+  private onFilterTextChanged = (filterText: string) => {
+    this.setState({ filterText })
+  }
+
+  private onSelectionChanged = (selectedBranch: Branch | null) => {
+    this.setState({ selectedBranch })
+  }
+
+  private renderBranch = (item: IBranchListItem, matches: IMatches) => {
+    return renderDefaultBranch(item, matches, this.props.currentBranch)
+  }
+
+  private onEnterPressed = (branch: Branch, source: ClickSource) => {
+    if (source.kind !== 'keyboard' || source.event.key !== 'Enter') {
+      return
+    }
+
+    source.event.preventDefault()
+
+    const { selectedBranch } = this.state
+
+    if (selectedBranch !== null && selectedBranch.name === branch.name) {
+      this.startCherryPick()
+    }
+  }
+
+  private canCherryPickOntoSelectedBranch() {
+    const { selectedBranch } = this.state
+    return selectedBranch !== null && !this.selectedBranchIsCurrentBranch()
+  }
+
+  private selectedBranchIsCurrentBranch() {
+    const { selectedBranch } = this.state
+    const currentBranch = this.props.currentBranch
+    return (
+      selectedBranch !== null &&
+      currentBranch !== null &&
+      selectedBranch.name === currentBranch.name
+    )
+  }
+
+  private renderOkButtonText() {
+    const okButtonText = 'Cherry pick commit'
+
+    const { selectedBranch } = this.state
+    if (selectedBranch !== null) {
+      return (
+        <>
+          {okButtonText} to <strong>{selectedBranch.name}</strong>…
+        </>
```

Comment:
What is this sorcery??? 😂 
I've been wrapping things in `div`s, how is this better?

Reply:
😄 I don't know if it is better per say. It is called a fragment. It is specifically for the purpose of needing JSX but not needing any attributes or specific html tag. https://medium.com/@yurachoi/react-js-fragments-175290da6435

Reply:
Benefit is it doesn't add to the DOM unnecessarily - had to think about it for a second to remember why I use it. 🙂